security/s2a/internal/crypter/ciphersuite.go

```diff
@@ -0,0 +1,109 @@
+package crypter
+
+import (
+	"crypto/sha256"
+	"golang.org/x/crypto/sha3"
+	s2a_proto "google.golang.org/grpc/security/s2a/internal/proto"
+	"hash"
+)
+
+// ciphersuite is the interface for a ciphersuite used in the S2A Half
+// Connection.
+type ciphersuite interface {
+	// keySize returns the key size in bytes.
+	keySize() int
+	// nonceSize returns the nonce size in bytes.
+	nonceSize() int
+	// trafficSecretSize returns the traffic secret size in bytes.
+	trafficSecretSize() int
+	// hashFunction returns the hash function for the ciphersuite.
+	hashFunction() func() hash.Hash
+	// aeadCrypter takes a key and creates an AEAD crypter for the ciphersuite
+	// using that key.
+	aeadCrypter(key []byte) (s2aAeadCrypter, error)
+}
+
+func newCiphersuite(ciphersuite s2a_proto.Ciphersuite) ciphersuite {
```

Comment:
Could we return `ciphersuite, err` instead? And in the case when the ciphersuite is unrecognized we can pass the error? I don't think we want to `panic` in the actual code.

Reply:
I think this got missed...